apkid/rules/elf/obfuscators.yara

```diff
@@ -187,3 +196,24 @@ rule firehash : obfuscator
   condition:
     elf.machine == elf.EM_ARM and all of them
 }
+
+rule avdobfuscator : obfuscator
+{
+  meta:
+    description = "AVDobfuscator"
+    url         = "https://github.com/andrivet/ADVobfuscator"
+    author      = "Eduardo Novella"
+    example     = "357f0c2ad6bf5cf60c671b090eab134251db63993f52aef512bde5bfa4a1b598"
+
+  strings:
+    $o1 = "ObfuscatedAddress"
```

Comment:
Is it possible to add some null bytes around these?

Reply:
Don't think so because these strings are used both for routines and exports.

**Strings**
```c
LOAD:000049CA   00000034    C   _ZNK17ObfuscatedAddressIPFiiiPciS0_S0_EE8originalEv
LOAD:00004A37   0000002B    C   _ZNK17ObfuscatedAddressIPFiPcEE8originalEv
LOAD:00004A62   0000002C    C   _ZNK17ObfuscatedAddressIPFvPciEE8originalEv
LOAD:00004A8E   0000002E    C   _ZNK17ObfuscatedAddressIPFvPcS0_EE8originalEv
LOAD:00004ABC   0000002A    C   _ZNK17ObfuscatedAddressIPFvvEE8originalEv
LOAD:00004AE6   0000003A    C   _Z14ObfuscatedCallI17ObfuscatedAddressIPFvvEEJEEvT_DpOT0_
LOAD:00004B20   0000002C    C   _ZNK17ObfuscatedAddressIPFiPviEE8originalEv
LOAD:00004B4C   0000002B    C   _ZNK17ObfuscatedAddressIPFvPcEE8originalEv
LOAD:00004CCF   00000032    C   _ZNK17ObfuscatedAddressIPFvP7_JNIEnvEE8originalEv
LOAD:00004D01   00000030    C   _ZNK17ObfuscatedAddressIPFvPcS0_iiEE8originalEv
LOAD:00004D31   0000002A    C   _ZNK17ObfuscatedAddressIPFvcEE8originalEv
LOAD:00004D5B   0000002D    C   _ZNK17ObfuscatedAddressIPFvPviiEE8originalEv
```

**Exports**
```c
Name    Address Ordinal
ObfuscatedAddress<int (*)(int,int,char *,int,char *,char *)>::original(void)    000151D0
ObfuscatedAddress<int (*)(char *)>::original(void)  00015220
ObfuscatedAddress<void (*)(char *,int)>::original(void) 00015228
ObfuscatedAddress<void (*)(char *,char *)>::original(void)  00015230
ObfuscatedAddress<void (*)(void)>::original(void)   00015238
ObfuscatedCall<ObfuscatedAddress<void (*)(void)>,>(ObfuscatedAddress<void (*)(void)>)   00015240
ObfuscatedAddress<int (*)(void *,int)>::original(void)  00015250
ObfuscatedAddress<void (*)(char *)>::original(void) 00015258
ObfuscatedAddress<void (*)(_JNIEnv *)>::original(void)  00016518
ObfuscatedAddress<void (*)(char *,char *,int,int)>::original(void)  00016520
ObfuscatedAddress<void (*)(char)>::original(void)   00016528
ObfuscatedAddress<void (*)(void *,int,int)>::original(void) 00016530
````
It would be good idea to check against these sumples if the AVDobfuscator signatures are always fixed. The strings there were taken from the source code @CalebFenton.

Perhaps we can remove a couple of them, that they are almost identical to another just for a character.